### ISSUE TITLE
Starter doesn't work on all supported react-native android versions (>=API16)

### ISSUE THREAD

**Issue by mikehardy** [invertase/react-native-firebase-starter#103]
The stock starter doesn't work on the full API skew supported by React-Native - they support down to API16 (as does my app) but this is set to a minimum API of API18

I'm not sure if there is a technical reason for this or not, but it appears Firebase itself supports those APIs just fine (I believe they target down to 14 or 15 still)

I was able to get a test app based on this starter working down to API16 though, with just a few changes - MultiDex support (which seems like it should be required for anything below API21, really?) and finishing the incomplete example app upgrade to react-native 0.58.x (the package.json depends on rc2 but hadn't done all the steps so gradle support lagged and there's a build error until 0.58.x is set up for real)

This project is valuable to me, I'd be happy to make a PR if you are interested?

**Comment by mikehardy** [comment]
@Salakar just checking interest before I move on from this area of work, can close otherwise

**Comment by dewelloper** [comment]
I can not build and sync same

**Comment by mikehardy** [comment]
@dewelloper please open a new issue if you need to, and include relevant details or no one will know what or why. The starter worked great for the APIs that it supported, and adb refuses to install the APK on ones it doesn't (with a clear message), so this was not a build + sync issue at all.

**Comment by dewelloper** [comment]
oke @mikehardy

**Comment by Salakar** [comment]
@mikehardy this is a limitation caused by notifications I believe or one of the other packages, can't quite remember off the top of my head - will looking into it but hoping this will no longer be the case with v6 - I want to add an Android API 16 CI stage nearer the release of v6.0.0 to make sure everything stays supported from min to max

**Comment by mikehardy** [comment]
I have the starter working down to API15 - the full react-native skew -  with correct MultiDex :-)

I'll see what I can do package up the changes - it's a bit hard to back-propagate changes in the starter because of the name change, but I understand why that exists because of firebase package/bundle name uniqueness needs

**Comment by Salakar** [comment]
It'd only be a short term thing so if it's a lot effort then I wouldn't worry too much.

----

The v6 starter just init's using RN's init template 🙃 so will be less conflicty

https://github.com/invertase/react-native-firebase/tree/master/packages/template

If I had a dollar though for everytime I mention v6 as the solution to something 😅
